Neos.Neos/Classes/FrontendRouting/EventSourcedFrontendNodeRoutePartHandler.php

```diff
@@ -223,7 +228,20 @@ public function matchWithParameters(&$requestPath, RouteParameters $parameters)
                 return false;
             }
         } catch (NodeNotFoundException $exception) {
-            // we silently swallow the Node Not Found case, as you'll see this in the server log if it interests you
+            // if there are fundamental errors with your dimension config (mismatch between database and settings), people only see a 404 - which is a problem because it
+            // is not possible for people to analyze this further. This is why in the error case, we check whether the root node exists in the expected DimensionSpacePoint, because
+            // this way, we can give some hints on what to do.
+            $this->securityContext->withoutAuthorizationChecks(function () use ($exception, $dimensionSpacePoint, $contentRepository) {
+                $sitesNodeAggregate = $contentRepository->getContentGraph(WorkspaceName::forLive())->findRootNodeAggregateByType(NodeTypeNameFactory::forSites());
```

Comment:
this code is now run everytime the frontend route part handler didnt match a route which might be valid as other handlers could continue. I like to introduce such debug helpers but id also think they should be nearly no overhead if everything is well... you have a comment in line 209 which states

> TODO validate dsp == complete (ContentDimensionZookeeper::getAllowedDimensionSubspace()->contains()...)

and id like to propose to do that first... validate that the `NodeInfo`'s dimensions would even exist of today? Does this make sense ... i might have to think harder on this one ^^ But then we can also live with that extra query - if you want add a route you generally put it infront of Neos.Neos so thats normally no bummer